src/EventSourcing/Domain/Aggregate/AggregateId.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSourcingWorkshop\EventSourcing\Domain\Aggregate;
+
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * @psalm-template AggregateType of Aggregate
+ * @psalm-immutable
+ */
+interface AggregateId
+{
+    /** @psalm-return non-empty-string */
+    public function toString(): string;
+
+    /** Note: if the aggregate ID is not a UUID, you may consider generating a UUID5 */
```

Comment:
The UUID5 part is clever, but this documentation does not explain why.

Reply:
Is there a reason why not going with uuid6?

Reply:
If the `AggregateId` is `"blah"` (not a UUID - happens often, for aggregates), then a UUID5 can be generated from it.

Internally, users are free to use `UUID6` for identifiers that **are** UUIDs.

Reply:
Removed entirely, for now: dd379ec18fca7f54ffda5eb234bf75c67d5caa02